### PR TITLE
Add single votes table

### PR DESCRIPTION
**Pull request by tsiegleauq**
Adds "SingleVoteTable", "Entitled Users", "general X to analog polls",
"enumber voting results", "user tokens"

### CODE REVIEW

**Comment by tsiegleauq** [comment]
This adds the client side of https://github.com/OpenSlides/OpenSlides/commit/4b13ff681e7073b174e5855e0f24b7208e981ca5

and also https://github.com/OpenSlides/OpenSlides/pull/6051 and https://github.com/OpenSlides/OpenSlides/pull/6050

Many fundamental stuff in voting has changed in os4. This cannot be tested since eVoting does not work yet. Nothing seems to break but I suppose the original author (Josh?) should review it due to the many changes that have happend in all poll objects

**Comment by FinnStutzenstein** [comment]
> This cannot be tested since eVoting does not work yet

To my knowledge, it should already work (I think except for delegations). If not it may be a bug

**Comment by tsiegleauq** [comment]
rebased. Reping.

File: client/src/app/site/polls/components/base-poll-detail.component.ts [hunk 1]
```diff
@@ -240,9 +250,43 @@ export abstract class BasePollDetailComponent extends BaseModelContextComponent
...
+                        ...entry,
+                        user: users.find(user => user.id === entry.user_id),
+                        voted_verbose: `voted:${entry.voted}`,
+                        vote_delegated_to: entry.vote_delegated_to_id
+                            ? users.find(user => user.id === entry.vote_delegated_to_id)
+                            : null
+                    });
+                }
+                this.entitledUsersObservable = from([entries]);
+            });
```
Comment (GabrielInTheWorld) [inline client/src/app/site/polls/components/base-poll-detail.component.ts:290]: The console logs the following error:
`entitled_users_at_stop` is undefined (at `base-poll-detail.component.ts` line 265) (Error: `TypeError: "this.poll.entitled_users_at_stop" is not iterable`)

File: client/src/app/site/motions/modules/motion-poll/motion-poll-detail/motion-poll-detail.component.html [hunk 2]
```diff
@@ -1,9 +1,9 @@
 <os-head-bar [goBack]="true" [nav]="false">
     <div class="title-slot">
-        <h2 *ngIf="poll">{{ 'Motion' | translate }} {{ poll.getContentObject().numberOrTitle }}</h2>
+        <h2 *ngIf="poll">{{ 'Motion' | translate }} {{ poll.motion.identifierOrTitle }}</h2>
```
Comment (GabrielInTheWorld) [inline client/src/app/site/motions/modules/motion-poll/motion-poll-detail/motion-poll-detail.component.html:3]: When I wanted to see my results, this error appears:
`TypeError: Cannot read property "identifierOrTitle" of undefined`

File: client/src/app/site/assignments/modules/assignment-poll/components/assignment-poll-detail/assignment-poll-detail.component.ts [hunk 3]
```diff
@@ -90,45 +93,44 @@ export class AssignmentPollDetailComponent extends BasePollDetailComponent {
...
-                        votes[userId].votes.push(this.voteValueToLabel(vote.value));
+                        const candidate_name =
+                            option.content_object?.getShortName() ?? this.translate.instant('Deleted user');
+                        votes[token].votes.push(`${candidate_name}: ${this.voteValueToLabel(vote.value)}`);
                     }
-                } else {
-                    votes[userId].votes.push(
-                        `${option.content_object.getShortName()}: ${this.voteValueToLabel(vote.value)}`
-                    );
                 }
```
Comment (GabrielInTheWorld) [inline client/src/app/site/assignments/modules/assignment-poll/components/assignment-poll-detail/assignment-poll-detail.component.ts:123]: Why do you changed this?

**Review by GabrielInTheWorld — CHANGES_REQUESTED** [review]
I have noticed, that before I can create a poll the `100%`-base cannot be choosed.

**Comment by tsiegleauq** [comment]
WIP
Added: User token on clients, entitled, and other changes I found in OS3.
autoupdates are still missing
some error regarding user_token
some error regarding entitled_users when voting stopped

File: client/src/app/site/motions/modules/motion-poll/motion-poll-detail/motion-poll-detail.component.html (hunk 2 above)
Comment (tsiegleauq) [reply client/src/app/site/motions/modules/motion-poll/motion-poll-detail/motion-poll-detail.component.html]: yep had too many changes here

File: client/src/app/site/polls/components/base-poll-detail.component.ts [hunk 4]
```diff
@@ -213,6 +209,24 @@ export abstract class BasePollDetailComponent extends BaseModelContextComponent
...
+                },
+                // {
+                //     idField: 'poll_voted' // not working
+                // },
+                // {
+                //     idField: {
+                //         templateIdField: 'poll_voted_$_ids',
+                //         templateValue: this.activeMeetingId?.toString()
+                //     }
+                // },
```
Comment (tsiegleauq) [inline client/src/app/site/polls/components/base-poll-detail.component.ts:229]: This is not getting me anywhere. The new request system still proves to be a foot gun.
Wither the relations are just wrong and "voted" is not what I expect it to be or requesting fields from the poll works different than all other models.

@FinnStutzenstein @GabrielInTheWorld

File: client/src/app/shared/models/poll/poll.ts [hunk 5]
```diff
@@ -35,6 +35,8 @@ export class Poll extends BaseDecimalModel<Poll> {
     public global_yes: boolean;
     public global_no: boolean;
     public global_abstain: boolean;
+    public is_pseudoanonymized: boolean;
+    public entitled_users_at_stop: EntitledUsersEntry[];
```
Comment (tsiegleauq) [inline client/src/app/shared/models/poll/poll.ts:39]: @jsangmeister @FinnStutzenstein I suppose this is not the correct place for `entitled_users_at_stop` in OS4 anymore? Where should it go?

**Comment by tsiegleauq** [comment]
User tokens seem to work fine.

single votes do not, some relations seem to be broken (again). Cannot get them from the autoupdate, tried several approaches
(@GabrielInTheWorld  @FinnStutzenstein )

the "entitled_users_at_stop" should not be too different from OS3. According to models.yml it is just a json fiield, therefore it should be possible to request it from the autoupdate service directly (poll repository, detailFieldset). Although the request to the autoupdate service seems fine, the field stays undefined.
(@jsangmeister)

File: client/src/app/site/polls/components/base-poll-detail.component.ts (hunk 4 above)
Comment (GabrielInTheWorld) [reply client/src/app/site/polls/components/base-poll-detail.component.ts]: I have committed an update to this PR. With this update you get the `voted`-users. (`poll.voted`)
You can read through this update and ask me, if you have questions.

File: client/src/app/shared/models/poll/poll.ts (hunk 5 above)
Comment (GabrielInTheWorld) [reply client/src/app/shared/models/poll/poll.ts:39]: I think, this is the right place.

**Comment by GabrielInTheWorld** [comment]
> the "entitled_users_at_stop" should not be too different from OS3. According to models.yml it is just a json fiield, therefore it should be possible to request it from the autoupdate service directly (poll repository, detailFieldset). Although the request to the autoupdate service seems fine, the field stays undefined.
> (@jsangmeister)

The field is initial `undefined` but will be filled after a while. Unfortunately at the moment the field is needed, it is `undefined`. The related auto-update comes later, so you have to wait for it.

**Comment by tsiegleauq** [comment]
rebased, fixed more bugs

File: client/src/app/site/polls/components/base-poll-detail.component.ts [hunk 6]
```diff
@@ -213,6 +192,10 @@ export abstract class BasePollDetailComponent extends BaseModelContextComponent
                 {
                     idField: 'content_object_id'
                 },
+                {
+                    idField: 'voted_ids',
+                    fieldset: 'shortName'
+                },
                 {
                     idField: 'option_ids',
                     follow: [{ idField: 'vote_ids' }, { idField: 'content_object_id' }]
```
Comment (GabrielInTheWorld) [inline client/src/app/site/polls/components/base-poll-detail.component.ts:201]: I have not noticed before, that you want to get the structure-level, the number and the vote-weight of a user.
To have this, you could re-write this line to:
```ts
                    follow: [
                        {
                            idField: 'vote_ids',
                            follow: [
                                {
                                    idField: 'user_id',
                                    fieldset: 'shortName',
                                    additionalFields: [
                                        { templateField: 'structure_level_$' },
                                        { templateField: 'vote_weight_$' },
                                        { templateField: 'number_$' }
                                    ]
                                }
                            ]
                        },
                        { idField: 'content_object_id' }
                    ]
```

**Review by GabrielInTheWorld — APPROVED** [review]
One hint. Everything else is good.

**Pull request merged**